docs/examples/hairpin_fitting/hairpin_unfolding.rst

```diff
@@ -0,0 +1,339 @@
+Sometimes, when fitting many unfolding events, the fit does not converge well when all data are fitted at once. If that happens, you can fix parameters from the first fit at this stage, for example by setting
+`fit["dna_handles/Lc"].fixed = True`. For this particular data set, the fit converges without fixing parameters. 
+We next set the initial guesses for the unfolded hairpin::
+
+    fit["dna_ss_hairpin/Lp"].lower_bound = 0.5  # in nanometers
+    fit["dna_ss_hairpin/Lp"].value = 1.5  # in nanometers
+    fit["dna_ss_hairpin/Lp"].upper_bound = 2.0  # in nanometers
+
+    fit["dna_ss_hairpin/Lc"].value = 0.02  # in microns
+    fit["dna_ss_hairpin/Lc"].lower_bound = 0.001  # in microns
+
+    fit["dna_ss_hairpin/St"].value = 500  # in pN
+    fit["dna_ss_hairpin/St"].upper_bound = 2000  # in pN
+
+Fit all the data and plot the result::
+
+   >>> fit.fit()
+   >>> print(fit.params)
+   Name                  Value       Unit      Fitted   Lower bound    Upper bound
+   --------------      ------------ --------  -------  -------------  -------------
+   dna_handles/Lp        34.7309    [nm]        True    30              100
+   dna_handles/Lc         1.75753   [micron]    True     0.00034        inf
+   dna_handles/St       904.301     [pN]        True     1              inf
+   kT                     4.11      [pN*nm]     False    3.77           8
+   dna_ss_hairpin/Lp      0.785985  [nm]        True     0.5            2
+   dna_ss_hairpin/Lc      0.0215748 [micron]    True     0.001          inf
+   dna_ss_hairpin/St   2000         [pN]        True     1              2000
+
+As can be seen from the table, most fitted parameters converge and have values in the expected range. However, the stretch modulus of the hairpin hits the upper bound of 2000 pN, indicating that this parameter did not converge.
+When observing that a parameter does not converge, it is important to go back to the fit and see how it can be improved. In this case, increasing the upper bound for `dna_ss_hairpin/St` does not visually change the fit and does not result in convergence;
+the stretch modulus of the DNA handles and the hairpin cannot be optimized at the same time. A solution would be to use the freely jointed chain, instead of the extensible freely jointed chain to fit the unfolded hairpin, which is equivalent to setting
+a very large value for `dna_ss_hairpin/St`::
+
+    fit["dna_ss_hairpin/St"].upper_bound = 100000 
+    fit["dna_ss_hairpin/St"].value = 100000  # in pN
+    fit["dna_ss_hairpin/St"].fixed = True 
+    fit.fit()
+
+Plot the result::
+
+    plt.figure()
+    fit[dna_handles_force].plot()
+    fit[dna_handles_and_hairpin_force].plot()
+    plt.xlabel(r"Distance ($\mu$m)")
+    plt.ylabel("Force (pN)")
+    Lc_hairpin = fit["dna_ss_hairpin/Lc"].value*1000
+    plt.title(f"Fitted hairpin length is {Lc_hairpin:0.1f} nm")
+
+.. image:: fit_all.png
+
+The expected contour length for the hairpin was ~17 nm and the fitted length is 21.7 nm. 
```

Comment:
It is always interesting to see just how sensitive this is. If I drop the data < 1.5 micron, I get `20.9`. If I just use the camera data `20.5`.